### PR TITLE
ASoC: SOF: imx8/imx8m: Replace sdev->private with sdev->pdata->hw_pdata

### PR DESCRIPTION
**Pull request by IulianOlaru249**
The correct way to save private data is to use sdev->pdata->hw_pdata.

Signed-off-by: Iulian Olaru <iulianolaru249@yahoo.com>

### CODE REVIEW

**Comment by dbaluta** [comment]
@IulianOlaru249  see checkpatch error.
When writing the commit message you need to have an empty line between commit subject and commit body.


```
ASoC: SOF: imx8/imx8m: Replace sdev->private with sdev->pdata->hw_pdata
<newline>
The correct way to save private data is to use sdev->pdata->hw_pdata.
```
You can set default editor for commit messages with the following command:

``` git config --global core.editor "vim" ``` 

Of course, you can set whatever editor you like instead of vim.

Also, please shorten the commit subject by replacing imx8/imx8m with imx.

e.g 

```ASoC: SOF: imx: Replace sdev->private with sdev->pdata->hw_pdata```

**Comment by dbaluta** [comment]
@ranj063 @plbossart after this commit is merged sdev->private is no longer used and it can be removed.

**Comment by plbossart** [comment]
Not sure what we see a SoundWire failure, will re-trigger tests.

**Comment by plbossart** [comment]
SOFCI TEST

File: sound/soc/sof/imx/imx8.c [hunk 1]
```diff
@@ -126,7 +126,7 @@ static struct imx_dsp_ops dsp_ops = {
 
 static int imx8_send_msg(struct snd_sof_dev *sdev, struct snd_sof_ipc_msg *msg)
 {
-	struct imx8_priv *priv = (struct imx8_priv *)sdev->private;
+	struct imx8_priv *priv = (struct imx8_priv *)sdev->pdata->hw_pdata;
```
Comment (lyakh) [inline sound/soc/sof/imx/imx8.c]: superfluous type-casts here and everywhere below, you have a good chance to remove them now :-)

**Pull request merged**